infer/src/base/Serialization.ml

```diff
@@ -61,7 +61,7 @@ let create_serializer (key : Key.t) : 'a serializer =
     let filename = DB.filename_to_string fname in
     PerfEvent.(
       log (fun logger -> log_begin_event logger ~name:("writing " ^ key.name) ~categories:["io"] ())) ;
-    Utils.with_intermediate_temp_file_out filename ~f:(fun outc ->
+    Utils.with_intermediate_temp_file_out ~retry:true filename ~f:(fun outc ->
```

Comment:
Right. Not having that ugly change would be a relief actually. It's not completely easy for us to retry in the next week or so. But I will remove the commit from the PR in the interim.

Note that this probably means that some of our changes on filenames are no longer relevant. Are there no more files names by the using procedures names?

Reply:
I think the only remaining use of these *as file names* (because we still use these strings as *keys*, but there there's no Windows limitation to consider) is the debug mode, which generates files in infer-out/captured/. In order for this debug mode to be helpful it would be good to keep the file prefix, hence why I was suggesting to lower the cutoff point on windows instead of removing the filenames altogether and keeping just the crc (well, I think that was one of the things I was suggesting).

Reply:
Ok, that's perfect. I simplify along those lines. (Although, unless you object, I will keep truncating the crc length, as it seems really wasteful.)

Reply:
Did you test if this is still needed? ^